### PR TITLE
feature: From Env Provider

### PR DESCRIPTION
**Pull request by james-milligan**
Simple environment variable based provider
No additional dependencies excluding the go-sdk

The purpose of this provider is to provide an example for use in a 'how to write a provider for the go-sdk' tutorial.

https://github.com/open-feature/golang-sdk-contrib/issues/17

### CODE REVIEW

File: providers/from-env/README.md
Comment (beeme1mr) [inline providers/from-env/README.md:60]: Maybe we should wait until https://github.com/open-feature/golang-sdk/issues/47 is addressed before merging this.

File: providers/from-env/README.md
Comment (beeme1mr) [inline providers/from-env/README.md:15]: How is targeting key being used?

Comment (james-milligan) [reply providers/from-env/README.md:15]: when set it must match the key provided in `EvaluationContext`, it might be an inappropriate use case, however I wanted to cover as many features as possible. This behaviour is briefly mentioned in the introduction however im happy to expand on it in the README

Comment (skyerus) [reply providers/from-env/README.md:15]: In the example below the targeting key user isn't being used right?

File: providers/from-env/pkg/provider.go
Comment (skyerus) [inline providers/from-env/pkg/provider.go:35]: We should probably hold off on merging these and update to remove EvaluationOptions once https://github.com/open-feature/golang-sdk/pull/48 has been merged

File: providers/from-env/README.md
Comment (james-milligan) [reply providers/from-env/README.md:15]: ill add an example for it now

File: providers/from-env/README.md
Comment (james-milligan) [reply providers/from-env/README.md]: now addressed, required changes have been made to this provider

File: providers/from-env/pkg/provider.go
Comment (james-milligan) [reply providers/from-env/pkg/provider.go]: these changes have been merged, examples and implementation have been updated :)

File: providers/from-env/README.md
Comment (skyerus) [inline providers/from-env/README.md:35]: minor but there's weird spacing here, can this be fixed?

Comment (james-milligan) [reply providers/from-env/README.md]: fixed :)

File: providers/from-env/README.md
Comment (beeme1mr) [inline providers/from-env/README.md:4]:
```suggestion
This provider uses a JSON evaluation for matching a flag `Variant` to a provided `EvaluationContext`. Each flag `Variant` contains a slice of `Conditions`, if all `Conditions` match then the flags value is returned. Each `Variant` is evaluated starting at index 0, therefore the first matching `Variant` is returned. Each variant also has a `TargetingKey`, when set it must match the `TargetingKey` provided in the `EvaluationContext` for the `Variant` to be returned.  
```

The first sentences already said this was a very simple implementation.

File: providers/from-env/README.md
Comment (beeme1mr) [inline providers/from-env/README.md:68]: This is be affected by your context change, right? If so, let's merge that first.

File: providers/from-env/pkg/provider.go
Comment (beeme1mr) [inline providers/from-env/pkg/provider.go:12]: Can these errors come from the SDK?

Comment (james-milligan) [reply providers/from-env/pkg/provider.go:12]: ive imported all available errors, and updated the style of the remaining errors / reasons to match

File: providers/from-env/README.md
Comment (james-milligan) [reply providers/from-env/README.md:68]: updated in latest commit

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**
